src/libcore/hash/sip.rs

```diff
@@ -193,6 +193,14 @@ impl Hasher for SipHasher {
     }
 
     #[inline]
+    fn delimit(&mut self, len: usize) {
+        // skip the first delimiter
+        if self.length > 0 {
+            self.write_usize(len);
```

Comment:
Actually I made a critical reading comprehension error earlier.  The discriminants are 0, both for user-defined types and the builtin `Option`; see http://is.gd/YjOcxM (playpen).